design-proposals/feature-lifecycle.md

```diff
@@ -0,0 +1,260 @@
+  from users to assure its usefulness and readiness for graduation.
+  If there is no confidence of usage or usefulness, it may remain in
+  this stage for some time.
+  However, the period in which a feature can remain in Beta is limited,
+  assuring features are not pilling up without control.
+
+  The API is considered stable with care no not break forward compatability.
+  This implies that only additional fields may be added during this stage
+  i.e. it is not allowed to change a field path/name or remove it.
+
+  The feature functionality is controlled using a Feature-Gate (FG) during
+  runtime. It must be specified for the feature to be active.
+
+- **GA**:
+  The feature graduated to general-availability (ga) and is now part of
+  the core features.
+
+  The feature functionality is no longer controlled by a FG.
+
+#### Discontinuation
+When and if a feature is targeted for deprecation and retirement,
+it needs to pass a deprecation process, depending on its current
+release stage (Alpha, Beta, GA).
+
+For more details, see [here](#deprecation).
+
+#### Release Stage Transition Table
+The following table summarized the different release stages with their
+transition requirements and restrictions.
+
+| Stage          | Period range    | F.Gate | Discontinuation Availability |
+|----------------|-----------------|--------|------------------------------|
+| Alpha          | 1 to 2 releases | YES    | Between **minor** releases   |
+| Beta           | 1 to 3 releases | YES    | Between **minor** releases   |
+| GA             | -               | NO     | Between **major** releases   |
+
+Through Alpha and Beta feature releases, a FG must be set in order
+for the feature to function.
+By default, no FG is specified, therefore the feature is by default disabled.
+
+If a feature is not able to transition to the next stage in the defined period,
+it should be discontinued automatically.
+
+> **Note**: Exceptions to the period range may apply
+> if 2/3 of active maintainers come to agreement to prolong
+> a specific feature.
+
+### Deprecation
+One reason for features to go through the Alpha and Beta stages,
+is the opportunity to examine its usefulness and adoption by users.
+Same goes with major releases that intentionally break backward
+compatability.
+
+Therefore, it is only natural that some features will not graduate
+between the stages, or will be found irrelevant after some time and be
+discontinued when transitioning between major releases.
+
+This section will mainly focus on discontinuation of features between
+minor project releases.
+Discontinuation between major project releases is somehow simpler,
+as it is common and agreement (see semver) that backward compatability
+can be broken between major releases. It is worth noting that when a
+new major release is introduced, the previous one is still maintained
+and supported, something that does not exist with minor releases.
```

Comment:
### Deprecation between major versions
    In semver, major versions are intended to break API compatbility.
    …